connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -238,7 +236,11 @@ public void send(K key, V value, org.apache.kafka.clients.producer.Callback call
         producer.send(new ProducerRecord<>(topic, key, value), callback);
     }
 
-
+    // package level visibility for testing only
+    void setTopicMetadataTimeoutMs(long timeoutMs) {
```

Comment:
might want to do this as an alternative, package-private constructor so we can at a minimum make `topicMetadataTimeoutMs` `final`.

If we don't change this, convention in Kafka is to not use `get`/`set` prefixes on the method name.